### PR TITLE
Upgrade stack in release/1.3 branch

### PR DESCRIPTION
**Pull request by peri044**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Comment by peri044** [comment]
https://github.com/pytorch/TensorRT/pull/1486 will disable conv3d patch issue

File: core/runtime/TRTEngine.cpp [hunk 1]
```diff
@@ -164,7 +164,7 @@ void TRTEngine::disable_profiling() {
 }
 
 void TRTEngine::dump_engine_layer_info_to_file(const std::string& path) {
-  auto inspector = cuda_engine->createEngineInspector();
+  auto inspector = make_trt(cuda_engine->createEngineInspector());
```
Comment (narendasan) [inline core/runtime/TRTEngine.cpp:167]: Just a note for later, should we be keeping this object around?

File: core/runtime/execute_engine.cpp [hunk 2]
```diff
@@ -121,36 +121,30 @@ std::vector<at::Tensor> execute_engine(std::vector<at::Tensor> inputs, c10::intr
...
     if (compiled_engine->profile_execution) {
       input_profiler_guard =
           std::make_unique<torch::autograd::profiler::RecordProfile>(compiled_engine->input_profile_path);
     }
-
-    contig_inputs.reserve(inputs.size());
-
     for (size_t i = 0; i < inputs.size(); i++) {
-      uint64_t pyt_idx = compiled_engine->in_binding_map[i];
+      std::string name = compiled_engine->exec_ctx->getEngine().getIOTensorName(i);
```
Comment (narendasan) [inline core/runtime/execute_engine.cpp:131]: @peri044 why arent we using input_binding_names here. TensorRT does not guarantee binding order. This would be potentially unsafe. 

I would expect that the workflow here would be to iterate not through `inputs.size()` but through the names in `input_binding_names`

File: core/runtime/execute_engine.cpp [hunk 3]
```diff
@@ -163,26 +157,27 @@ std::vector<at::Tensor> execute_engine(std::vector<at::Tensor> inputs, c10::intr
 
     for (size_t o = inputs.size(); o < (compiled_engine->num_io.first + compiled_engine->num_io.second); o++) {
```
Comment (narendasan) [inline core/runtime/execute_engine.cpp:158]: Same thing here, use output_binding_names to pull out the TensorRT binding index / address

File: core/runtime/execute_engine.cpp (hunk 2 above)
Comment (narendasan) [reply core/runtime/execute_engine.cpp:131]: this should be `std::string name = compiled_engine->input_binding_names[i]`

File: py/torch_tensorrt/fx/converters/convolution.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 import numpy as np
 import tensorrt as trt
 import torch
+import warnings
```
Comment (narendasan) [inline py/torch_tensorrt/fx/converters/convolution.py:5]: Other parts of the codebase have been using logging for this sort of role.

File: py/torch_tensorrt/fx/test/passes/test_fuse_permute_linear_trt.py [hunk 5]
```diff
@@ -52,29 +52,31 @@ def forward(self, x):
             apply_passes=[fuse_permute_linear],
         )
 
-    def test_multi_fuse_permute_linear(self):
-        """
-        Fusion when permute output is shared by multiple linears
-        """
+    # TODO: The following test has been disabled due to a bug in TRT 8.5.1.7
```
Comment (yinghai) [inline py/torch_tensorrt/fx/test/passes/test_fuse_permute_linear_trt.py:55]: Can we just use unittest skip?

**Review by yinghai — CHANGES_REQUESTED** [review]
Please address the warning and unittest skip comment. Otherwise LGTM.

**Review by yinghai — APPROVED** [review]
lgtm
